### DISCUSSION TITLE
Can't get bearer token from request in C# function

### DISCUSSION THREAD

**Discussion by myers-gh1328** [Azure/static-web-apps#719] [Q&A]
In the JS example for custom role assignment at https://github.com/staticwebdev/roles-function/blob/main/api/GetRoles/index.js the bearer token for calling graph is retrieved by setting a const user = req.body and then getting the token from user.accessToken 

The request body in C# is a stream that does not contain anything. What is the C# equivalent to the js example code?

**Comment by mkarmark** [comment]
The format of accessing fields in the request body is the same for this context as for normal HTTP trigger C# functions (https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=csharp).

The following code snippet can be used to access the access token for example:
``` c#
string requestBody = string.Empty;
using (StreamReader streamReader =  new  StreamReader(req.Body))
{
     requestBody = await streamReader.ReadToEndAsync();
}
		
dynamic data = JsonConvert.DeserializeObject(requestBody);
string accessToken = data?.accessToken;
```

**Reply by myers-gh1328** [reply]
This does not work. The request body is always empty in my function. I am using the azure managed function, and running it through the Static Web App CLI emulator.

**Reply by simonaco** [reply]
@mmyers299 we have an open issue in the static web apps CLI to enable local debugging of these roles [Debug the roles function using the local auth emulator](https://github.com/Azure/static-web-apps-cli/issues/364). What happens when you run your application in production? Are your able to see the user token?
